Ray.jl/src/object_store.jl

```diff
@@ -20,7 +20,7 @@ if run in an `@async` task.
 If the task that generated the `ObjectID` failed with a Julia exception, the
 captured exception will be thrown on `get`.
 """
-get(oid::ray_jll.ObjectIDAllocated) = _get(take!(ray_jll.get(oid)))
+get(oid::ray_jll.ObjectIDAllocated; timeout_ms::Int=-1) = _get(take!(ray_jll.get(oid, timeout_ms)))
```

Comment:
Having `take!` return a `nothing` and `deserialize_from_bytes(::Nothing)` seems off. What about:
```suggestion
function get(oid::ray_jll.ObjectIDAllocated; timeout_ms::Int=-1)
    buffer = ray_jll.get(oid, timeout_ms)
    return isnull(buffer[]) ? nothing : _get(take!(buffer))
end
```

Reply:
Having `ray_jll.get` return a `nullptr` is sensible if we hit the timeout. The interface for `take!` however is to read a `Buffer` and return the data as bytes. I could see having the `isnull` check inside `take!` if we could have the function return a `UInt8[]` but as a non-null `Buffer` could store an empty vector (note any serialized data will be non-empty due to the serialization header) we shouldn't do this.

The only way we'd see segfaults is if we pass in a `nullptr` to `take!`. As these low level functions shouldn't be used by end users I'm not seeing an issue with checking for `nullptr` after the `ray_jll.get` call.